### PR TITLE
Feature/003 rx adapter

### PR DESCRIPTION
**Pull request by JiceDessaint**

### CODE REVIEW

**Comment by raphrel** [comment]
le dossier RxModernAVPlayer dans le projet XCode est rouge car referencé sur "ModernAVPlayer/New Group" qui n'existe pas.

File: ModernAVPlayer.podspec [hunk 1]
```diff
@@ -9,8 +9,20 @@ ModernAVPlayer is an ongoing project that aims to create a more usable audio vid
   s.license          = { :type => 'MIT', :file => 'LICENSE' }
   s.author           = { 'raphael ankierman' => 'raphrel@gmail.com' }
   s.source           = { :git => 'https://github.com/raphrel/ModernAVPlayer.git', :tag => s.version.to_s }
+  s.default_subspec  = 'Core'
 
   s.ios.deployment_target = '10.0'
   s.swift_version = '4.0'
-  s.source_files = 'Sources/**/*'
+
+s.subspec 'Core' do |core|
```
Comment (raphrel) [inline ModernAVPlayer.podspec]: probleme d'indentation

File: Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    
+    public static func registerKnownImplementations() {
+        self.register { RxPlayerContextDelegateProxy(playerContext: $0) }
+    }
+
+    // MARK: - Proxy Subjects
+    
+    internal lazy var stateSubject = PublishSubject<ModernAVPlayer.PlayerState>()
+    internal lazy var currentTimeSubject = PublishSubject<Double?>()
```
[the remark this answers is not in the document]
Comment (JiceDessaint) [reply Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift]: IMHO, 'subject' is not really a 'suffix'. It's part of the variable identity.
'state' and 'stateSubject' or 'statePublisher' are not the same thing. The first one suggest some kind of variable, whereas the other ones clearly exhibits some kind of "pushing data".

Subject is more than just a type here, not like 'itemsCount' and 'itemsCountInteger' for example.

Don't you think ? :)

File: Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    
+    public func playerContext(_ context: ConcretePlayerContext, currentItemUrl: URL?) {
+        currentItemUrlSubject.onNext(currentItemUrl)
+    }
+
+    // MARK: - Deinit
+
+    deinit {
+        self.stateSubject.on(.completed)
```
[the remark this answers is not in the document]
Comment (JiceDessaint) [reply Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift]: yep

File: Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                                           PlayerContextDelegate {
+    
+    // MARK: - Initialization
+    
+    public init(playerContext: ConcretePlayerContext) {
+        super.init(parentObject: playerContext, delegateProxy: RxPlayerContextDelegateProxy.self)
+    }
+    
+    public static func registerKnownImplementations() {
+        self.register { RxPlayerContextDelegateProxy(playerContext: $0) }
```
[the remark this answers is not in the document]
Comment (JiceDessaint) [reply Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift]: done

File: Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    // MARK: - Proxy Subjects
+    
+    internal lazy var stateSubject = PublishSubject<ModernAVPlayer.PlayerState>()
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift]: ok so just remove `internal` keywords

File: Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/RxModernAVPlayer/PlayerContextDelegateProxy.swift]: ok

**Comment by JiceDessaint** [comment]
Duplication with #11

**Pull request closed without merging**